digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/lane/LaneDao.scala

```diff
@@ -0,0 +1,454 @@
+
+    sqlu"""INSERT INTO LANE (id, lane_code, created_date, created_by, municipality_code)
+          VALUES ($laneId, ${newIncomeLane.laneCode}, sysdate, ${username}, ${newIncomeLane.municipalityCode} )
+    """.execute
+
+    laneId
+  }
+
+
+  def insertLaneAttributes(laneId: Long, laneProp: LaneProperty, username: String): Long = {
+
+    val laneAttributeId = Sequences.nextPrimaryKeySeqValue
+    val laneAttrValue = if( laneProp.values.isEmpty ) ""
+                        else laneProp.values.head.value.toString
+
+    sqlu"""INSERT INTO lane_attribute (id, lane_id, name, value, created_date, created_by)
+            VALUES( $laneAttributeId, $laneId, ${laneProp.publicId}, ${laneAttrValue}, sysdate, $username)
+    """.execute
+
+    laneAttributeId
+  }
+
+
+  def deleteEntryLane( laneId: Long ): Unit = {
+
+    val laneCode = sql"""SELECT lane_code FROM LANE WHERE id = $laneId""".as[Long].first
+
+    if ( MAIN_LANES.contains(laneCode) )
+      throw new IllegalArgumentException("Cannot Delete a main lane!")
+
+    sqlu"""DELETE FROM LANE_ATTRIBUTE WHERE lane_id = $laneId""".execute
+    sqlu"""DELETE FROM LANE_LINK WHERE lane_id = $laneId""".execute
+    sqlu"""DELETE FROM LANE WHERE id = $laneId""".execute
+
+    sqlu"""DELETE FROM LANE_POSITION
+           WHERE id = (SELECT lane_position_id
+                        FROM LANE_LINK
+                        WHERE lane_id = $laneId )""".execute
+  }
+
+
+  def updateEntryLane( lane: PersistedLane, username: String ): Long = {
+
+    updateLane(lane, username)
+    updateLanePosition(lane, username)
+
+    lane.attributes match {
+      case props: LanePropertiesValues =>
+        props.properties.filterNot( _.publicId == "lane_code" )
+                        .foreach( attr => updateLaneAttributes(lane.id, attr, username) )
+
+      case _ => None
+    }
+
+    lane.id
+  }
+
+  def updateLane(lane: PersistedLane, username: String ): Unit = {
+
+    val oldLaneCode = sql"""Select lane_code FROM LANE WHERE id = ${lane.id}""".as[Int].first
+
+    if ( MAIN_LANES.contains(oldLaneCode) && oldLaneCode != lane.laneCode )
+      throw new IllegalArgumentException("Cannot change the code of main lane!")
+
+    sqlu"""UPDATE LANE
+          SET LANE_CODE = ${lane.laneCode}, MODIFIED_BY = $username, MODIFIED_DATE = sysdate, MUNICIPALITY_CODE = ${lane.municipalityCode}
+          WHERE id = ${lane.id}
+    """.execute
+  }
+
+  def updateLanePosition(lane: PersistedLane, username: String ): Unit = {
+    sqlu"""UPDATE LANE_POSITION
+           SET SIDE_CODE = ${lane.sideCode}, START_MEASURE = ${lane.startMeasure}, END_MEASURE = ${lane.endMeasure}, LINK_ID = ${lane.linkId}
+          WHERE ID = (SELECT LANE_POSITION_ID FROM LANE_LINK WHERE LANE_ID = ${lane.id})
+     """.execute
+  }
+
+  def updateLaneAttributes(laneId: Long, props: LaneProperty, username: String ): Unit = {
+
+    val finalValue = if (props.values.isEmpty) "Null"
+                     else props.values.head.value.toString
```

Comment:
If we don't have the value why insert a Null and not just ignore the query?

Reply:
To avoid erros in extrations and in other future operations when for some reason a new properties is inserted and no one create an insert or update to add this new property.
Even for some reason the LaneProperty.values comes empty instead an error of Null.head beeing thown, we place null value in DB